File: apps/ios/Barns/BarnsApp/Data/Repositories/MockAuthRepository.swift
/// In-memory mock auth. No persistence, no network, no real credentials.
actor MockAuthRepository: AuthRepository {
    private var signedInUser: User?
    private let mockUser = User(id: "mock-user-001", displayName: "Guest")

    func currentUser() async throws -> User? {
        signedInUser
    }

    func signInWithMockAccount() async throws -> User {
        signedInUser = mockUser
        return mockUser
    }

    func signOut() async throws {
        signedInUser = nil
    }
}

File: apps/android/app/src/main/java/com/barns/app/data/repository/MockAuthRepository.kt
package com.barns.app.data.repository

import com.barns.app.domain.model.User
import com.barns.app.domain.repository.AuthRepository

/**
 * In-memory mock auth. No persistence, no network, no real credentials.
 */
class MockAuthRepository : AuthRepository {
    private val lock = Any()
    private var signedInUser: User? = null
    private val mockUser = User(id = "mock-user-001", displayName = "Guest")

    override suspend fun currentUser(): User? =
        synchronized(lock) { signedInUser }

    override suspend fun signInWithMockAccount(): User =
        synchronized(lock) {
            signedInUser = mockUser
            mockUser
        }

    override suspend fun signOut() {
        synchronized(lock) {
            signedInUser = null
        }
    }
}
